python/tests/test_version.py

```diff
@@ -14,16 +14,12 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-import re
 
-from pyiceberg import __version__
+from packaging.version import Version
 
-VERSION_REGEX = re.compile(r"^\d+.\d+.\d+(.(dev|rc)\d+)?$")
+from pyiceberg import __version__
 
 
 def test_version_format():
-    # should be in the format of 0.14.0 or 0.14.0.dev0
-    assert VERSION_REGEX.search(__version__)
-
-    # RCs should work as well
-    assert VERSION_REGEX.search("0.1.0.rc0")
+    # We should be able to parse the version
+    assert Version(__version__)
```

Comment:
This seems reasonable to me. I don't think that the rc number should be included in release candidate versions, though. That was the main issue.